### ISSUE TITLE
the following yaml file 1e5 is regareded as string instead of float

### ISSUE THREAD

**Issue by smilesun** [marrlab/DomainLab#429]
```
mode: grid

output_dir: zoutput/benchmarks/benchmark_fbopt

num_param_samples: 8
sampling_seed: 0
startseed: 0
endseed: 2

test_domains:
  - 3
  - 0


domainlab_args:
  task: mnistcolor10
  tr_d: [1, 2]
  dmem: False
  lr: 0.001
  epos: 3
  es: 5
  bs: 64
  nname: conv_bn_pool_2
  san_check: True
  exp_shoulder_clip: 10
  mu_clip: 1e4
  coeff_ma: 0.5


Shared params:
  ini_setpoint_ratio:
    min: 0.5
    max: 0.99
    num: 5
    step: 0.05
    distribution: uniform

  k_i_gain:
    min: 0.0001
    max: 0.01
    num: 2
    step: 0.0001
    distribution: uniform

# Test fbopt with different hyperparameter configurations

jigen_fbopt:
  aname: jigen
  trainer: fbopt

  shared:
    - ini_setpoint_raio
```

**Comment by e-dorigatti** [comment]
https://stackoverflow.com/questions/30458977/yaml-loads-5e-6-as-string-and-not-a-number

**Comment by smilesun** [comment]
One has to switch to fbopt branch to run this yaml file.

**Comment by e-dorigatti** [comment]
this is a YAML bug, you could first try to use a different parser such as [ruamel.yaml](https://pypi.org/project/ruamel.yaml/)

**Comment by Car-la-F** [comment]
@e-dorigatti 
Thanks for the suggestion. I tried it out with the ruamel.yaml package and it would resolve this bug (and many other we always had when reading in the yaml files). 

@smilesun we would need to add in this new package to the requirements and change the reading of the yaml files everywhere in the package. If you find this is a good solution I can start working on this.

**Comment by smilesun** [comment]
if it is a good solution that would get us free from those existing and future bugs, i guess we can use it.

**Comment by smilesun** [comment]
previously we have forced the datatype at hyper-parameter sampling/search by using float()
`https://github.com/marrlab/DomainLab/blob/b974cac67e4d803f44eab264ded25d48a7bc4038/domainlab/utils/hyperparameter_gridsearch.py#L49`

**Comment by smilesun** [comment]
in master:  
`
./run_benchmark_standalone.sh examples/benchmark/benchmark_mnist_hard.yaml`

result in 

```
RuleException:
AttributeError in file /home/sunxd/domainlab/domainlab/exp_protocol/benchmark.smk, line 101:
'DataFrame' object has no attribute 'append'
  File "/home/sunxd/domainlab/domainlab/exp_protocol/benchmark.smk", line 101, in __rule_parameter_sampling
  File "/home/sunxd/domainlab/domainlab/utils/hyperparameter_sampling.py", line 434, in sample_hyperparameters
  File "/home/sunxd/domainlab/domainlab/utils/hyperparameter_sampling.py", line 351, in sample_task
  File "/home/sunxd/domainlab/domainlab/utils/hyperparameter_sampling.py", line 326, in sample_task_only_shared
  File "/home/sunxd/anaconda3/lib/python3.9/site-packages/pandas/core/generic.py", line 6204, in __getattr__
  File "/home/sunxd/anaconda3/lib/python3.9/concurrent/futures/thread.py", line 58, in run
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2023-10-23T122320.735144.snakemake.log
```

**Comment by smilesun** [comment]
line 319 of hyperparameter_grid_search.py

**Comment by smilesun** [comment]
trying a fix here
https://github.com/marrlab/DomainLab/pull/585
and here
https://github.com/marrlab/DomainLab/pull/584

**Comment by smilesun** [comment]
fixed in [1aebf84](https://github.com/marrlab/DomainLab/commit/1aebf84ea2c61dc3820a06ac6f89ec93ebc9a674)
and commit b4b9ce66bad58310e3cae17564c218d84a4bd660
https://github.com/marrlab/DomainLab/commit/b4b9ce66bad58310e3cae17564c218d84a4bd660
